### DISCUSSION TITLE
how to add files in workpath?

### DISCUSSION THREAD

**Discussion by changxiaoju** [deepmodeling/dpgen#436] [Q&A] [answered]
are there any method to add files used in fp calculations? like a command in machine.json to put a file in work path everytime.

**Comment by AnguseZhang** [comment, marked as answer]
Do you mean an extra input file, but not generated by DP-GEN in each task path? Could you please specify the need more clearly?

**Reply by changxiaoju** [reply]
yes, like if fp using vdw-DF, i need to have a file vdw_kernel.bindat in directory, how to put it properly

**Reply by AnguseZhang** [reply]
Currently there is no support for this, I can add this function. Do you have other related needs?

**Reply by changxiaoju** [reply]
Thank you, nothing else for the moment

**Reply by AnguseZhang** [reply]
This has implemented in https://github.com/AnguseZhang/dpgen/tree/fix_481 .
Please search "user_forward_files" for instructions of this function.
